common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1257 @@
+	// keep a count of bytes left to be written increase as we read, decrease as we write.
+	// Then call the callback if we hit EOF on read and bytesToWrite === 0
+	let bytesToWrite = 0;
+	let doneReading = false;
+	Ti.Stream.pump(srcStream, obj => {
+		if (obj.bytesProcessed === -1) {
+			doneReading = true;
+		} else {
+			bytesToWrite += obj.bytesProcessed;
+		}
+
+		Ti.Stream.write(destStream, obj.buffer, 0, obj.buffer.length, writeObj => {
+			// If not EOF/error, decrease # of bytes left to write
+			if (writeObj.bytesProcessed !== -1) {
+				bytesToWrite -= writeObj.bytesProcessed;
+			}
+
+			// if we're done reading and we've written as many bytes as we read, call the callback
+			if (doneReading && bytesToWrite === 0) {
+				callback();
+			}
+		});
+	}, COPY_FILE_CHUNK_SIZE, true);
+};
+
+/**
+ * @param {string|Buffer|URL} src source filename to copy
+ * @param {string|Buffer|URL} dest destination filename of the copy operation
+ * @param {number} [flags=0] modifiers for copy operation
+ */
+fs.copyFileSync = function (src, dest, flags = 0) {
+	const srcFile = Ti.Filesystem.getFile(src);
+	if (flags === fs.constants.COPYFILE_EXCL) {
+		// TODO: fail if dest already exists!
+	}
+	if (!srcFile.copy(dest)) {
+		throw new Error(); // FIXME: What error should we give?
+	}
+};
+
+// TODO: fs.createReadStream(path, options)
+// TODO: fs.createWriteStream(path, options)
+
+/**
+ * @callback existsCallback
+ * @param {boolean} exists - whether path exists
+ */
+
+/**
+ * @param {string} path path to check
+ * @param {existsCallback} callback callback function
+ * @returns {void}
+ */
+fs.exists = function (path, callback) {
+	setTimeout(() => {
+		callback(fs.existsSync(path));
+	}, 1);
+};
+
+/**
+ * @param {string} path path to check
+ * @returns {boolean} whether a file or directory exists at that path
+ */
+fs.existsSync = function (path) {
+	try {
+		fs.accessSync(path);
+		return true;
+	} catch (e) {
+		return false;
+	}
+};
+
+fs.fchmod = (fd, mode, callback) => setTimeout(callback, 1);
+fs.fchmodSync = NO_OP;
+
+fs.fchown = (fd, uid, gid, callback) => setTimeout(callback, 1);
+fs.fchownSync = NO_OP;
+
+fs.fdatasync = (fd, callback) => callback();
+fs.fdatasyncSync = () => {};
+
+/**
+ * @param {integer} fd file descriptor
+ * @param {object} [options] options
+ * @param {boolean} [options.bigint] whether stat values should be bigint
+ * @param {function} callback async callback function
+ */
+fs.fstat = (fd, options, callback) => {
+	setTimeout(() => {
+		if (typeof options === 'function') {
+			callback = options;
+			options = {};
+		}
+
+		let stats;
+		try {
+			stats = fs.fstatSync(fd, options);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback(null, stats);
+	}, 1);
+};
+/**
+ * @param {integer} fd file descriptor
+ * @param {object} [options] options
+ * @param {boolean} [options.bigint] whether stat values should be bigint
+ * @returns {fs.Stats} stats for file descriptor
+ */
+fs.fstatSync = (fd, options) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 408;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 408 – 'options' is defined but never used. Allowed unused args must match /^_.+/. (no-unused-vars)